plugins/module_utils/client.py

```diff
@@ -81,11 +90,31 @@ def get(self, type_name: str) -> ResourceType:
         return ResourceType(json.loads(result["Schema"]))
 
 
+class AwsBotocoreError(Exception):
+    def __init__(self, exc, msg):
+        self.exc = exc
+        self.msg = msg
+        super().__init__(self)
+
+
 class AwsClient:
     def __init__(self, **kwargs: Any) -> None:
```

Comment:
I would actually prefer to change the constructor to just accept an already created boto3 session object. Using dependency injection here should make testing easier and it would generally be cleaner to delegate boto3 client config to something else.

My recommendation would be that this change to something like:
```python
def __init__(self, session=None, discoverer=None) -> None:
```

Reply:
Now that I'm looking at your Azure client PR I'm realizing we'll need some kind of CloudClient interface, so we can't have something specific to boto like a session as an argument.